### PR TITLE
Feat: Use new car parking lots data source

### LINKED ISSUE #604: Use new car and bike parking lots map layer datasource
Right now the bike parking and car parking layers use two different sources and two different classes for fetching the data.

However, we moved and combined the data source in the backend and there is now only a single data source for both types of parking.

Just like with the stops, if one of the layers is activated in the layer switcher, you need to fetch the data for both and then filter out the parking lots that you do/don't want to display client-side.

The data format has changed quite a bit but all the information that you need for both icons and side panels should still be there. If it isn't, please get in touch.

A JSON representation of the new data is this:

```
      // this is a bicycle parking station
      {
        "name": "Bicycle parking",
        "lon": 7.7667109000000005,
        "lat": 48.578502300000004,
        "imageUrl": null,
        "tags": [
          "osm:amenity=bicycle_parking"
        ],
        "anyCarPlaces": false
      },
     // this is a statically derived park and ride parking lot
      {
        "name": "P+R",
        "lon": 8.497435600000001,
        "lat": 47.947423150000006,
        "imageUrl": null,
        "tags": [
          "osm:amenity=parking"
        ],
        "anyCarPlaces": true
      },
     // this is a dynamically created and updated parking lot that receives live updates about available spaces
     {
        "name": "Nufringer Tor",
        "lon": 8.870266,
        "lat": 48.597865,
        "capacity": {
          "carSpaces": 242
        },
        "availability": {
          "carSpaces": 142
        },
        "imageUrl": null,
        "tags": [
          "lot_type:Parkhaus",
          "address:Nufringer Tor",
          "forecast:false"
        ],
        "anyCarPlaces": true
      },
```      

The base URL to fetch the layer information from is 
```
API_URL/routing/v1/router/vectorTiles/parking/
```

If you want to preview the data in the layer, you can use a combination of curl and protoc.
```
curl https://api.dev.stadtnavi.eu/routing/v1/router/vectorTiles/parking/17/68763/45237.pbf | protoc --decode_raw
```

Fixes https://github.com/stadtnavi/digitransit-ui/issues/445

### PR DESCRIPTION
**Pull request by andreashelms**
## Proposed Changes

  - Use new car parking lots data source

## Review

  - Read and verify the code changes
  - Test the functionality by running the UI locally with all popular browsers available in your platform

closes #604

### CODE REVIEW

File: app/component/map/tile-layer/DynamicParkingLots.js [hunk 1]
```diff
@@ -98,7 +122,11 @@ class DynamicParkingLots {
       );
     }
 
-    const icon = DynamicParkingLots.getIcon(properties.lot_type);
+    const tags = properties.tags.split(',').reduce((newTags, tag) => {
+      return { ...newTags, [tag.split(':')[0]]: tag.split(':')[1] };
+    }, {});
```
Comment (andreashelms) [inline app/component/map/tile-layer/DynamicParkingLots.js:127]: If backend returns tags as JSON-encoded string, we can remove this here.

Comment (leonardehrenfried) [reply app/component/map/tile-layer/DynamicParkingLots.js:127]: Sorry, I have bad news: https://github.com/stadtnavi/digitransit-ui/issues/604#issuecomment-863854611

Comment (andreashelms) [reply app/component/map/tile-layer/DynamicParkingLots.js:127]: OK, then I will make sure the application won't break if tags does not contain key value pairs.

Comment (andreashelms) [reply app/component/map/tile-layer/DynamicParkingLots.js:127]: Works as desired. I think we can merge.

**Comment by leonardehrenfried** [comment]
When you activate both bike and car park layers are the tiles downloaded once or twice? Does the browser realise that it's the same URL?

**Comment by andreashelms** [comment]
> When you activate both bike and car park layers are the tiles downloaded once or twice? Does the browser realise that it's the same URL?

You are right, if there is no caching defined in service worker (I think that's the case) the tiles are downloaded twice with this implementation. I will try to change that.

**Comment by leonardehrenfried** [comment]
If it makes it easier for the frontend I can set the appropriate caching headers but I think I already do that.

**Comment by leonardehrenfried** [comment]
```
curl -I https://api.dev.stadtnavi.eu/routing/v1/router/vectorTiles/parking/17/68763/45237.pbf
HTTP/2 200 
server: nginx/1.14.2
date: Fri, 18 Jun 2021 08:50:55 GMT
content-type: application/x-protobuf
content-length: 2294
cache-control: no-transform, max-age=60
access-control-allow-credentials: false
```

**Comment by andreashelms** [comment]
image

I just realized browser already serves data from cache because of the configured cache-control (0B transfered via network with second request). So I think the implementation is fine.

**Comment by leonardehrenfried** [comment]
Well done!

**Pull request merged**